### PR TITLE
Render runtime type checks for var_args bifs

### PR DESCRIPTION
**Pull request by awelzel**
Currently, Zeek disables any static type checking for var_arg bifs. However, the generated preamble for var_args bifs assume that typed positional arguments are correctly typed and blindly calls the type converters on them. This easily triggers abort()s at runtime currently when a script mistakenly uses the wrong types for var_arg bifs. For example, calling publish_rr() with a port instead of a string causes a hard-abort with Zeek 5.0.8.

    $ zeek -e 'Cluster::publish_rr(Cluster::Pool(), 80/tcp)'
    fatal error in <no location>: Val::CONVERTER (port/string) (80/tcp)
    Aborted (core dumped)

Extend bifcl so that for var_arg functions and the types that bifcl understands, we render a runtime type check and explicit early return to avoid the abort(). For any/other types, the implementer of the bif continuous to be responsible for type checking.

This isn't solving the var_args situation generally, but avoids some ad-hoc fixes trickling in current bif implementations.

Some references:

https://github.com/zeek/zeek/issues/1523
https://github.com/zeek/zeek/issues/2425
https://github.com/zeek/zeek/issues/2935
https://github.com/zeek/zeek/pull/2950

### CODE REVIEW

**Comment by awelzel** [comment]
Excerpt from the `publish_rr_bif()` function generated with this patch:
```
        zeek::Val* pool = (zeek::Val*) ((*BiF_ARGS)[0].get());
                {
                // Runtime type check for key argument
                zeek::TypeTag __tag = (*BiF_ARGS)[1]->GetType()->Tag();
                if ( __tag != TYPE_STRING )
                        {
                        zeek::emit_builtin_error(zeek::util::fmt("expected type string for key, got %s", zeek::type_name(__tag)));
                        return nullptr;
                        }
                }
        zeek::StringVal* key = (zeek::StringVal*) ((*BiF_ARGS)[1].get()->AsStringVal());
```

File: builtin-func.y [hunk 1]
```diff
@@ -708,7 +708,7 @@ body_start:	TOK_LPB c_code_begin
 				fprintf(fp_func_def, "\tif ( %s->size() != %d )\n", arg_list_name, argc);
 				fprintf(fp_func_def, "\t\t{\n");
 				fprintf(fp_func_def,
-					"\t\treporter->Error(\"%s() takes exactly %d argument(s)\");\n",
+					"\t\tzeek::emit_builtin_error(zeek::util::fmt(\"%s() takes exactly %d argument(s)\"));\n",
```
Comment (timwoj) [inline builtin-func.y]: Why the change from reporter to builtin here?

Comment (awelzel) [reply builtin-func.y]: Because the emit_builtin_error() is including location information which the plain reporter doesn't. I didn't have that initially and some of the baselines over in Zeek lost the location info and I found that sad.

**Review by timwoj — APPROVED** [review]
